### PR TITLE
mrf24j40: Don't use netdev_ieee802154_t for link layer address

### PR DESCRIPTION
**Pull request by bergzand**
### Contribution description

This PR modifies the behaviour of the mrf24j40 driver to not propagate the link layer address to the netdev_ieee802154_t struct. This helps the goal of separating the netdev and the ieee802154_t layer by another bit. The end goal is to remove the short and long address from the netdev_ieee802154_t struct and match the behaviour of the ethernet drivers.

The performance impact of this change should be negligible, the NETOPT_ADDRESS(_LONG) call is only used on initialization and by the ifconfig command.

### Testing procedure

Test whether:

 -  ifconfig still shows the correct link layer addresses. These should not have changed between this PR and master.
 - ping still works. This to verify that the link layer address is also properly configured in the radio, if the ping echo/reply fails, the address filter of the radio might be configured in the wrong endianness.

### Issues/PRs references

related to #10401, part of #7736

### CODE REVIEW

**Comment by bergzand** [comment]
The last commit replaces a number of shifts by `byteorder.h` types. This shaves of 32B of flash usage and should make these operations endianness safe.

**Comment by bergzand** [comment]
blocking until #10537 is merged

**Comment by miri64** [comment]
#10537 got merged into master. Please rebase.

**Comment by bergzand** [comment]
Rebased!

File: drivers/mrf24j40/mrf24j40_netdev.c [hunk 1]
```diff
@@ -174,6 +174,26 @@ static int _get(netdev_t *netdev, netopt_t opt, void *val, size_t max_len)
...
+
+        case NETOPT_ADDRESS_LONG:
+            if (max_len < sizeof(uint64_t)) {
+                res = -EOVERFLOW;
+            }
+            else {
+                *(uint64_t*)val = mrf24j40_get_addr_long(dev);
+                res = sizeof(uint64_t);
+            }
+            break;
```
Comment (miri64) [inline drivers/mrf24j40/mrf24j40_netdev.c]: Ok this part is different, but it makes sense and is sane ;-)

**Comment by gschorcht** [comment]
work as expected

**Comment by miri64** [comment]
@bergzand please squash

**Comment by bergzand** [comment]
squashed!

**Pull request merged**